JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesSparql.java

```diff
@@ -41,27 +42,113 @@
 	public static final String ns_kb = "https://www.theworldavatar.com/kg/ontotimeseries/";
-	
+	public static final String ns_time = "http://www.w3.org/2006/time#";
+
 	// Prefixes
 	private static final Prefix prefix_ontology = SparqlBuilder.prefix("ts", iri(ns_ontology));
 	private static final Prefix prefix_kb = SparqlBuilder.prefix("kb", iri(ns_kb));
-	
-	// RDF type
-    private static final Iri TimeSeries = prefix_ontology.iri("TimeSeries");
-    
-    // Relationships
-    private static final Iri hasTimeSeries = prefix_ontology.iri("hasTimeSeries");
-    private static final Iri hasRDB = prefix_ontology.iri("hasRDB");
-    private static final Iri hasTimeUnit = prefix_ontology.iri("hasTimeUnit");
-
-    // Fields for class specific exceptions
+	private static final Prefix prefix_time = SparqlBuilder.prefix("time", iri(ns_time));
+
+	// Relationships
+	private static final Iri hasTimeSeries = prefix_ontology.iri("hasTimeSeries");
+	private static final Iri hasRDB = prefix_ontology.iri("hasRDB");
+	private static final Iri hasTimeUnit = prefix_ontology.iri("hasTimeUnit");
+	private static final Iri hasAveragingPeriod = prefix_ontology.iri("hasAveragingPeriod");
+	private static final Iri numericDuration = prefix_time.iri("numericDuration");
+	private static final Iri unitType = prefix_time.iri("unitType");
+
+	// Fields for class specific exceptions
 	private final String exceptionPrefix = this.getClass().getSimpleName() + ": ";
 
-    /**
+	//Time Series Type
+	public static final String AverageTypeString = ns_ontology+"AverageTimeSeries";
+	public static final String StepwiseCumulativeTypeString = ns_ontology+"StepwiseCumulativeTimeSeries";
+	public static final String CumulativeTotalTypeString = ns_ontology+"CumulativeTotalTimeSeries";
+	public static final String InstantaneousTypeString = ns_ontology+"InstantaneousTimeSeries";
+	public static final String TimeSeriesTypeString = ns_ontology+"TimeSeries";
+
+	// RDF type
+	public static final Iri TimeSeries = iri(TimeSeriesTypeString);
+	public static final Iri StepwiseCumulativeTimeSeries = iri(StepwiseCumulativeTypeString);
+	public static final Iri CumulativeTotalTimeSeries = iri(CumulativeTotalTypeString);
+	public static final Iri AverageTimeSeries = iri(AverageTypeString);
+	public static final Iri InstantaneousTimeSeries = iri(InstantaneousTypeString);
+	private static final Iri Duration = prefix_time.iri("Duration");
+
+	//Timeseries IRI prefix
+	public static final String AverageIRIString = ns_ontology + "AverageTimeseries";
+	public static final String StepwiseCumulativeIRIString = ns_ontology + "StepwiseCumulativeTimeseries";
+	public static final String CumulativeTotalIRIString = ns_ontology + "CumulativeTotalTimeseries";
+	public static final String InstantaneousIRIString = ns_ontology + "InstantaneousTimeseries";
+	public static final String TimeSeriesIRIString = ns_ontology+"Timeseries";
```

Comment:
either use ns_kb here or combine ns_ontology and ns_kb into a single constant since they are the same

Reply:
resolved